buildconfig/CMake/Eigen.cmake

```diff
@@ -8,7 +8,8 @@ option(USE_SYSTEM_EIGEN "Use the system installed Eigen - v${eigen_version}?" OF
 
 if(USE_SYSTEM_EIGEN)
   message(STATUS "Using system Eigen")
-  find_package(Eigen ${eigen_version} EXACT REQUIRED)
+  find_package(Eigen3 ${eigen_version} EXACT REQUIRED
```

Comment:
To make this useful to redhat7, we'd have to change it to a minimum version starting at 3.2.5. Dropping the version requirement might not be a big deal as build servers should pick up errors in our usage and most systems will use the external projects.